google-pubsub/1.0/modules/ROOT/pages/google-pubsub-connector-reference.adoc

```diff
@@ -0,0 +1,641 @@
+| Project ID a| String |  Globally unique identifier for your project. |  |
+| Snapshot Name a| String |  Project-wide unique identifier for a snapshot. |  | x
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  #[payload] |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+=== Output
+
+[%autowidth.spread]
+|===
+|Type |<<Snapshot>>
+|===
+
+=== For Configurations
+
+* <<Config>>
+
+=== Throws
+
+* PUBSUB:BAD_GATEWAY
+* PUBSUB:CANCELLED
+* PUBSUB:CONNECTIVITY
+* PUBSUB:DEADLINE_EXCEEDED
+* PUBSUB:FAILED_PRECONDITION
+* PUBSUB:INVALID_ARGUMENT
+* PUBSUB:NOT_FOUND
+* PUBSUB:PERMISSION_DENIED
+* PUBSUB:RETRY_EXHAUSTED
+* PUBSUB:UNAVAILABLE
+
+
+[[GetSnapshotList]]
+== Get Snapshot List
+`<pubsub:get-snapshot-list>`
+
+Retrieves a list of existing snapshots from a selected project.
+
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Project Id a| String |  Globally unique identifier for your project. |  |
+| Config Ref a| ConfigurationProvider |  Name of the configuration used to execute this component. |  | x
+| Streaming Strategy a| * <<RepeatableInMemoryIterable>>
+* <<RepeatableFileStoreIterable>>
+* non-repeatable-iterable |  Configures how Mule processes streams. The default is to use repeatable streams. |  |
+| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
+| Target Value a| String |  Expression that evaluates the operation’s output. The outcome of the expression is stored in the *Target Variable* field. |  #[payload] |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+=== Output
+
+[%autowidth.spread]
+|===
+|Type |Array of <<Snapshot>>
+|===
+
+=== For Configurations
+
+* <<Config>>
+
+=== Throws
+
+* PUBSUB:BAD_GATEWAY
+* PUBSUB:CANCELLED
+* PUBSUB:DEADLINE_EXCEEDED
+* PUBSUB:FAILED_PRECONDITION
+* PUBSUB:INVALID_ARGUMENT
+* PUBSUB:NOT_FOUND
+* PUBSUB:PERMISSION_DENIED
+* PUBSUB:UNAVAILABLE
+
+
+[[PatchSnapshot]]
+== Patch Snapshot
+`<pubsub:patch-snapshot>`
+
+
+Updates an existing snapshot.
+
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Snapshot a| Any |  Snapshot object with updated fields. |  #[payload] |
+| Update Mask a| String |  Indicates which fields in the provided snapshot to update. Must be specified and not empty. This is a comma-separated list of fully qualified names of fields. |  | x
```

Comment:
You can remove the second sentence since the field is required.

This is a comma-separated list of fully qualified names of fields. > This is a comma-separated list of fully qualified field names.

Reply:
Removed the second sentence. Kept "fully qualified names" since that is the actual term name